### PR TITLE
Alexej/eng 14 integration test the examples within ci pipeline

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Running `poetry run bash scripts/test-coverage-xml.sh` now runs all unit and integration tests. Its possible to run `poetry run bash scripts/test-coverage-xml.sh unit` or `poetry run bash scripts/test-coverage-xml.sh integration` to specify which tests to run. Will add this to the docs once its accepted.

### CODE REVIEW

File: tests/integration/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    tmp_path = tmp_path_factory.mktemp("tmp")
+    examples_path = tmp_path / "zenml_examples"
+    source_path = Path("examples")
+    shutil.copytree(source_path, examples_path)
+    yield examples_path
+    shutil.rmtree(str(examples_path))
+
+
+def test_run_quickstart(examples_dir: Path):
+    """"""
```
Comment (schustmi) [inline tests/integration/test_examples.py:37]: I think we should either have a small docstring here or just remove the quotes entirely (same for the other functions)

**Review by schustmi — CHANGES_REQUESTED** [review]
Looks good, just two small changes:
One is a separate comment, and `tests/unit/__init__.py ` is missing a license

File: tests/integration/test_examples.py (hunk 1 above)
Comment (AlexejPenner) [reply tests/integration/test_examples.py:37]: I must have gotten distracted when I wanted to write them

**Review by schustmi — APPROVED** [review]
Nice, I really like the new checks if certain steps are cached etc!

**Pull request merged**